homeassistant/components/http/security_filter.py

```diff
@@ -59,34 +63,48 @@ async def security_filter_middleware(
         """Process request and block commonly known exploit attempts."""
         path_with_query_string = f"{request.path}?{request.query_string}"
 
+        base_msg = None
         for unsafe_byte in UNSAFE_URL_BYTES:
             if unsafe_byte in path_with_query_string:
                 if unsafe_byte in request.query_string:
-                    _LOGGER.warning(
-                        "Filtered a request with unsafe byte query string: %s",
-                        request.raw_path,
-                    )
-                    raise HTTPBadRequest
-                _LOGGER.warning(
-                    "Filtered a request with an unsafe byte in path: %s",
-                    request.raw_path,
-                )
-                raise HTTPBadRequest
-
-        if FILTERS.search(_recursive_unquote(path_with_query_string)):
-            # Check the full path with query string first, if its
-            # a hit, than check just the query string to give a more
-            # specific warning.
-            if FILTERS.search(_recursive_unquote(request.query_string)):
-                _LOGGER.warning(
-                    "Filtered a request with a potential harmful query string: %s",
-                    request.raw_path,
+                    base_msg = "Filtered a request from {ip} with unsafe byte query string: {path}"
+                    break
+
+                base_msg = (
+                    "Filtered a request from {ip} with an unsafe byte in path: {path}"
                 )
-                raise HTTPBadRequest
+                break
+
+        if base_msg is None:
+            if FILTERS.search(_recursive_unquote(path_with_query_string)):
+                # Check the full path with query string first, if its
+                # a hit, than check just the query string to give a more
+                # specific warning.
+                if FILTERS.search(_recursive_unquote(request.query_string)):
+                    base_msg = "Filtered a request from {ip} with a potential harmful query string: {path}"
+                else:
+                    base_msg = (
+                        "Filtered a potential harmful request from {ip} to: {path}"
+                    )
 
-            _LOGGER.warning(
-                "Filtered a potential harmful request to: %s", request.raw_path
+        if base_msg is not None:
+            # Circular import with websocket_api
+            # pylint: disable=import-outside-toplevel
+            from ipaddress import ip_address
+
+            from homeassistant.components import persistent_notification
+
+            ip_address_ = ip_address(request.remote)  # type: ignore[arg-type]
+            hass = request.app[KEY_HASS]
+            msg = base_msg.format(ip=ip_address_, path=request.path)
+            _LOGGER.warning(msg)
+            persistent_notification.async_create(
```

Comment:
We generally don't accept PRs to add new persistent notifications

Reply:
Happy to remove this if a deal breaker - I find it useful so I don't have to check the logs daily